File: le1/plp/expressions1/Programa.java
package le1.plp.expressions1;

import le1.plp.expressions1.expression.Expressao;
import le1.plp.expressions1.expression.Valor;

public class Programa{

	private Expressao exp;

	public Valor executar() {
		Valor result = exp.avaliar();
		System.out.println(result);
		return  result;		
 	}

	public boolean checaTipo() {
		return exp.checaTipo();
 	}

	public Programa(Expressao exp){
		this.exp = exp;
	}

	public Expressao getExpressao() {
		return exp;
	}

}

File: Le1 - Python/le1/programa.py
from __future__ import annotations

from le1.expression.expressao import Expressao
from le1.expression.valor import Valor


class Programa:
    def __init__(self, exp: Expressao) -> None:
        self._exp = exp

    def executar(self) -> Valor:
        result = self._exp.avaliar()
        print(result)
        return result

    def checaTipo(self) -> bool:
        return self._exp.checaTipo()

    def getExpressao(self) -> Expressao:
        return self._exp